x/community/abci_test.go

```diff
@@ -0,0 +1,119 @@
+	suite.Keeper = suite.App.GetCommunityKeeper()
+
+	// Set up x/mint and x/kavadist gen state
+	mintGen := minttypes.DefaultGenesisState()
+	mintGen.Params.InflationMax = sdk.NewDecWithPrec(595, 3)
+	mintGen.Params.InflationMin = sdk.NewDecWithPrec(595, 3)
+	kavadistGen := kavadisttypes.DefaultGenesisState()
+	kavadistGen.Params.Active = true
+	appCodec := tApp.AppCodec()
+	suite.App.InitializeFromGenesisStates(
+		app.GenesisState{minttypes.ModuleName: appCodec.MustMarshalJSON(mintGen)},
+		app.GenesisState{kavadisttypes.ModuleName: appCodec.MustMarshalJSON(kavadistGen)},
+	)
+}
+
+func TestABCITestSuite(t *testing.T) {
+	suite.Run(t, new(ABCITestSuite))
+}
+
+func (suite *ABCITestSuite) TestBeginBlockerDisableInflationUpgrade() {
+	validateUpgradedState := func() {
+		mintParams := suite.App.GetMintKeeper().GetParams(suite.Ctx)
+		suite.Equal(sdk.ZeroDec(), mintParams.InflationMax)
+		suite.Equal(sdk.ZeroDec(), mintParams.InflationMin)
+
+		kavadistParams := suite.App.GetKavadistKeeper().GetParams(suite.Ctx)
+		suite.False(kavadistParams.Active)
+	}
+
+	suite.Run("starts disable inflation upgrade on vanilla chain", func() {
+		suite.SetupTest()
+		community.BeginBlocker(suite.Ctx, suite.Keeper)
+		validateUpgradedState()
+	})
+
+	suite.Run("starts disable inflation upgrade when past upgrade time", func() {
+		suite.SetupTest()
+
+		suite.setUpgradeTimeFromNow(time.Hour * -1)
+		community.BeginBlocker(suite.Ctx, suite.Keeper)
+		validateUpgradedState()
+	})
+
+	suite.Run("don't upgrade if upgrade time is in the future", func() {
+		suite.SetupTest()
+
+		suite.setUpgradeTimeFromNow(time.Hour * 1)
+		community.BeginBlocker(suite.Ctx, suite.Keeper)
+
+		mintParams := suite.App.GetMintKeeper().GetParams(suite.Ctx)
+		suite.NotEqual(sdk.ZeroDec(), mintParams.InflationMax)
+		suite.NotEqual(sdk.ZeroDec(), mintParams.InflationMin)
+
+		kavadistParams := suite.App.GetKavadistKeeper().GetParams(suite.Ctx)
+		suite.True(kavadistParams.Active)
+	})
+
+	suite.Run("don't run upgrade if already upgraded", func() {
+		suite.SetupTest()
+
+		community.BeginBlocker(suite.Ctx, suite.Keeper)
+		validateUpgradedState()
+
+		kavadistParams := suite.App.GetKavadistKeeper().GetParams(suite.Ctx)
+		kavadistParams.Active = true
+		suite.App.GetKavadistKeeper().SetParams(suite.Ctx, kavadistParams)
+
+		suite.Ctx = suite.Ctx.WithBlockTime(suite.Ctx.BlockTime().Add(time.Minute * 6))
+		community.BeginBlocker(suite.Ctx, suite.Keeper)
+
+		mintParams := suite.App.GetMintKeeper().GetParams(suite.Ctx)
+		suite.Equal(sdk.ZeroDec(), mintParams.InflationMax)
+		suite.Equal(sdk.ZeroDec(), mintParams.InflationMin)
+
+		kavadistParams = suite.App.GetKavadistKeeper().GetParams(suite.Ctx)
+		suite.True(kavadistParams.Active)
+	})
+}
+
+func (suite *ABCITestSuite) setUpgradeTimeFromNow(t time.Duration) {
+	suite.Keeper.SetParams(
+		suite.Ctx,
+		types.Params{UpgradeTimeDisableInflation: suite.Ctx.BlockTime().Add(t)},
```

Comment:
Should be updated to include the new `StakingRewardsPerSecond` param, test is panicking with it being nil